src/odemis/gui/cont/settings.py

```diff
@@ -719,3 +723,200 @@ def __init__(self, tab_panel, tab_data):
+
+        # Show the dialog and check whether is was accepted or cancelled
+        if dialog.ShowModal() != wx.ID_OK:
+            return
+
+        # get selected path + filename and update default directory
+        self._calib_path = dialog.GetDirectory()
+        path = dialog.GetPath()
+        filename = dialog.GetFilename()
+
+        # read file
+        with open(path, 'rb') as csvfile:
+            calibFile = csv.reader(csvfile, delimiter=':')
+
+            try:
+                triggerDelay_dict = calibration.get_time_range_to_trigger_delay(calibFile,
+                                                                                self.streak_unit.timeRange.choices,
+                                                                                self.streak_delay.triggerDelay.range)
+            except ValueError as error:
+                self._onUpdateTriggerDelayGUI(ERROR, filename)  # update txt displayed in GUI
+                logging.error(error)
+                return
+
+        # update the MD
+        self.streak_delay.updateMetadata({model.MD_TIME_RANGE_TO_DELAY: triggerDelay_dict})
+
+        # update triggerDelay shown in GUI
+        cur_timeRange = self.streak_unit.timeRange.value
+        # find the corresponding trigger delay
+        tr2d = self.streak_delay.getMetadata()[model.MD_TIME_RANGE_TO_DELAY]
+        key = odemis.util.find_closest(cur_timeRange, tr2d.keys())
+        # Note: no need to check almost_equal again as we do that already when loading the file
+        self.streak_delay.triggerDelay.value = tr2d[key]  # set the new value
+
+        self._onUpdateTriggerDelayGUI(LOAD, filename)  # update txt displayed in GUI
+
+    def _onSaveCalibFile(self, event):
+        """
+        Saves a calibration file (*csv) containing the time range and the corresponding trigger delay
+        for streak camera calibration.
+        """
+        logging.debug("Save trigger delay calibration file for temporal acquisition.")
+
+        dialog = wx.FileDialog(self.panel,
+                               message="Choose a filename and destination to save the calibration file. "
+                                       "It is advisory to include the SEM voltage into the filename.",
+                               defaultDir=self._calib_path,
+                               defaultFile="",
+                               style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
+                               wildcard="csv files (*.csv)|*.csv")
+
+        # Show the dialog and check whether is was accepted or cancelled
+        if dialog.ShowModal() != wx.ID_OK:
+            return
+
+        # get selected path + filename and update default directory
+        self._calib_path = dialog.GetDirectory()
+        path = dialog.GetPath()
+        filename = dialog.GetFilename()
+
+        # check if filename is provided with the correct extension
+        if filename.split(".")[-1] != "csv":
```

Comment:
Please use `os.path.splitext()`. Your code works in most cases fine, but the official function of Python makes it more explicit, and handles some of the corner cases (ex, the user selects as filename "csv").